### PR TITLE
[Merged by Bors] - feat(Data/Nat/Defs): pow_eq_self_iff

### PR DESCRIPTION
**Pull request by rwst**
For `a > 1`, `a ^ b = a` iff `b = 1`. Lemma needed by #14049.

Co-authored-by: Ruben Van de Velde <65514131+Ruben-VandeVelde@users.noreply.github.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Defs.lean [hunk 1]
```diff
@@ -770,6 +770,10 @@ protected lemma pow_right_injective (ha : 2 ≤ a) : Injective (a ^ ·) :=by
   | 0 => by simp
   | n + 1 => by rw [Nat.pow_succ, mul_eq_zero, Nat.pow_eq_zero]; omega
 
+/-- For `a > 1`, `a ^ b = a` iff `b = 1`. -/
+lemma pow_eq_self_iff (a b : ℕ) (ha : 2 ≤ a) : a ^ b = a ↔ b = 1 :=
```
Comment (tb65536) [inline Mathlib/Data/Nat/Defs.lean:774]: I think `1 < a` might be preferrable to `2 ≤ a` (although they are definitionally equal).

Comment (rwst) [reply Mathlib/Data/Nat/Defs.lean]: Done.

File: Mathlib/Data/Nat/Defs.lean [hunk 2]
```diff
@@ -770,6 +770,10 @@ protected lemma pow_right_injective (ha : 2 ≤ a) : Injective (a ^ ·) :=by
   | 0 => by simp
   | n + 1 => by rw [Nat.pow_succ, mul_eq_zero, Nat.pow_eq_zero]; omega
 
+/-- For `a > 1`, `a ^ b = a` iff `b = 1`. -/
+lemma pow_eq_self_iff (a b : ℕ) (ha : 1 < a) : a ^ b = a ↔ b = 1 :=
```
Comment (tb65536) [inline Mathlib/Data/Nat/Defs.lean:774]: I think `a` should be implicit here, and perhaps also `b` since this is an `iff` lemma.

Comment (rwst) [reply Mathlib/Data/Nat/Defs.lean]: Hope I did this correctly.

**Comment by tb65536** [comment]
maintainer merge

**Comment by rwst** [comment]
Thanks!

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**